apps/verticalhome/js/pinned_apps_navigation.js

```diff
@@ -13,274 +13,375 @@
         }
+      }
 
-        if (toRemoveOutFocus) {
-          toRemoveOutFocus.classList.remove('out-focus');
-        }
+      this.refresh();
+    },
+
+    currentIndex: function currentIndex(){
+      var currIndex = -1;
+      for(var i = 0; i < this.elements.length; i++){
+        if(this.elements[i].classList.contains('selected')){
+            currIndex = i;
+          }
       }
+      return currIndex;
+    },
 
-      if (this.preventDef){
-        e.preventDefault();
+    scrollNum: function scrollNum(_scrollElemNum){
+      var scrollElemNum = _scrollElemNum || 1;
+      var ElsLength = this.elements.length;
+
+      if( (scrollElemNum > ElsLength) ||
+        (scrollElemNum < 0 && scrollElemNum<(ElsLength)*-1)
+      ){
+        scrollElemNum = scrollElemNum % ElsLength;
       }
 
-      var topStyle = null;
+      var step = parseInt(window.getComputedStyle(this.elements[0])
+        .getPropertyValue('height'));
 
-      switch(e.key) {
-        case 'ArrowUp':
+      var top_pos = window.getComputedStyle(this.elemList,null)
+        .getPropertyValue('top');
 
-          preVerticalNavigation();
-          document.getElementById('clock').classList.add('not-visible');
+      var fix = 110;
 
-          if (this.elemList.dataset.scrollup) {
-            this.elemList.style.top = this.elemList.dataset.scrollup;
-          } else {
-            this.elemList.style.top = '-6rem';
-            this.elemList.dataset.scrollup = this.elemList.style.top;
-          }
+      if(scrollElemNum<0){
+        fix = 0;
+      }
 
-          var deltaElemsToStartScroll = 3;
+      this.elemList.classList.add('animation');
+      this.elemList.style.top =
+        parseInt(top_pos)-((step+this.stepKoef)*scrollElemNum)+fix+'px';
 
-          this.elements[this.selectedElemIndex].classList.remove('selected');
-          if (this.selectedElemIndex >= 0 && this.selectedElemIndex < 4) {
+      return true;
+    },
 
-            var cycledElem = deltaElemsToStartScroll - this.selectedElemIndex;
-            cycleElements(cycledElem);
-            this.selectedElemIndex += cycledElem;
+    afterAnimation: function afterAnimation(){
+      var pinnedAppsNavElList = app.pinnedAppsNavigation.elemList;
+      pinnedAppsNavElList.classList.remove('animation');
+      pinnedAppsNavElList.style.top = app.pinnedAppsNavigation.defTopval;
 
-            this.selectedElemIndex--;
-            this.elements[this.selectedElemIndex].classList.add('selected');
+       return true;
+    },
+    /**
+     * Handle event keydown
+     * @param  {[Object]} e [default object]
+     * @return {[nothing]}   [nothing]
+     */
+    handleEvent: function(e) {
+      var mainscreen = document.getElementById('main-screen');
+      this.personalizeMode = mainscreen.classList.contains('personalize_mode');
+      this.rearrangeMode = mainscreen.classList.contains('rearrange_mode');
 
-          } else {
+      if (this.preventDef) {
+        e.preventDefault();
+      }
```

Comment:
"You really can't prevent default until you've checked the keycode and know you're going to handle the key. This will break the volume keys, for example." :copyright: **(David)**

I think it is unnecessary. It would be removed.